internal/registration/internal/application/command_handler.go

```diff
@@ -0,0 +1,87 @@
+package application
+
+import (
+	"GoEdu/internal/registration/internal/domain"
+	"context"
+	"fmt"
+)
+
+type CommandHandler interface {
+	Handle(context context.Context, command Command) (CommandResult, error)
+}
+
+type RegisterNewUserCommandHandler struct {
+	hasher     PasswordHasher
+	verifier   UniqueEmailVerifier
+	repository UserRegistrationRepository
+}
+
+func (r *RegisterNewUserCommandHandler) Handle(context context.Context, command Command) (CommandResult, error) {
+	regNewUserCommand := command.(RegisterNewUserCommand)
+
+	user, err := domain.RegisterNewUser(
+		regNewUserCommand.FirstName,
+		regNewUserCommand.LastName,
+		regNewUserCommand.Email,
+		r.hasher.Hash(regNewUserCommand.Password),
+	)
+
```

Comment:
It's not necessary to place a space over here because it's an atomic block of code: call a function and check an error.